### PR TITLE
syncing changes back from enterprise

### PR DESCRIPTION
**Pull request by rboyer**
Changes reflect 2022-04-05.

### CODE REVIEW

File: agent/delegate_mock_test.go [hunk 1]
```diff
@@ -47,9 +47,9 @@ func (m *delegateMock) RemoveFailedNode(node string, prune bool, entMeta *struct
 	return m.Called(node, prune, entMeta).Error(0)
 }
 
-func (m *delegateMock) ResolveTokenAndDefaultMeta(token string, entMeta *structs.EnterpriseMeta, authzContext *acl.AuthorizerContext) (acl.Authorizer, error) {
+func (m *delegateMock) ResolveTokenAndDefaultMeta(token string, entMeta *structs.EnterpriseMeta, authzContext *acl.AuthorizerContext) (consul.ACLResolveResult, error) {
 	ret := m.Called(token, entMeta, authzContext)
-	return ret.Get(0).(acl.Authorizer), ret.Error(1)
+	return ret.Get(0).(consul.ACLResolveResult), ret.Error(1)
```
Comment (eculver) [inline agent/delegate_mock_test.go:52]: It's difficult to see how this is correct but looking at other references to `consul.ACLResolveResult` and the fact that it's a test, I will assume it's good.

**Review by eculver — APPROVED** [review]
LGTM.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/626070.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit d06183ba7f516bedef3b7c717f2f553dc25ed495 onto `release/1.12.x` succeeded!

**Pull request merged**